cocos/ui/UIPageView.cpp

```diff
@@ -145,12 +145,24 @@ void PageView::setCurPageIndex( ssize_t index )
 
 void PageView::setCurrentPageIndex(ssize_t index)
 {
+	auto preindex = MAX(0, getCurrentPageIndex());
+		
     jumpToItem(index, Vec2::ANCHOR_MIDDLE, Vec2::ANCHOR_MIDDLE);
+
+	if (preindex != index)
+	{
+		pageTurningEvent();
```

Comment:
I'm also curious about this change, do we really need fire  a turning event in `setCurrentPageIndex` ?